src/test/kotlin/org/rust/cargo/toolchain/CargoTest.kt

```diff
@@ -70,15 +77,43 @@ class CargoTest : RsTestBase() {
     """)
 
     fun `test adds nightly channel`() = checkCommandLine(
-        cargo.toColoredCommandLine(CargoCommandLine("run", wd, listOf("--release", "--", "foo"), channel = RustChannel.NIGHTLY)), """
+        cargo.toColoredCommandLine(project, CargoCommandLine("run", wd, listOf("--release", "--", "foo"), channel = RustChannel.NIGHTLY)), """
         cmd: /usr/bin/cargo +nightly run --color=always --release -- foo
         env: RUST_BACKTRACE=short, TERM=ansi
         """, """
         cmd: C:/usr/bin/cargo.exe +nightly run --color=always --release -- foo
         env: RUST_BACKTRACE=short, TERM=ansi
     """)
 
-    private fun checkCommandLine(cmd: GeneralCommandLine, expected: String, expectedWin: String) {
+    @MockRustcVersion("1.36.0")
+    fun `test adds --offline option`() {
+        project.rustSettings.modify { it.useOffline = true }
+        checkCommandLine(cargo.toColoredCommandLine(project, CargoCommandLine("run", wd, listOf("--release", "--", "foo"))), """
+            cmd: /usr/bin/cargo --offline run --color=always --release -- foo
+            env: RUST_BACKTRACE=short, TERM=ansi
+            """, """
+            cmd: C:/usr/bin/cargo.exe --offline run --color=always --release -- foo
+            env: RUST_BACKTRACE=short, TERM=ansi
+        """)
+    }
+
+    @MockRustcVersion("1.35.0-nightly")
+    fun `test adds -Zoffline`() {
+        project.rustSettings.modify { it.useOffline = true }
```

Comment:
Let's restore initial value at the end of tests.
I think it may look the following
```kotlin
withOfflineMode {
    // some actions
}
```
where `withOfflineMode` sets `useOffline` to true, runs action and at the end restores `useOffline` initial value.
It should help us to avoid unexpected test results because of shared project state between some tests